src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -540,9 +556,140 @@ private List<GeneratorTemplate> getModElementGeneratorTemplatesList(ModElement e
 			}
 		}
 
+		Objects.requireNonNull(getModElementListTemplates(element, performFSTasks, generatableElement)).forEach(gtl -> {
+			for (ListTemplate generatorTemplate : gtl.templates().keySet()) {
+				for (int i = 0; i < gtl.listData().size(); i++) {
+					if (gtl.templates().get(generatorTemplate).get(i) || !performFSTasks) {
+						files.add(new ListTemplate(gtl.processTokens(generatorTemplate, i),
+								generatorTemplate.getTemplateIdentificator(), gtl, i,
+								generatorTemplate.getTemplateData()));
+					}
+				}
+			}
+		});
+
 		return new ArrayList<>(files);
 	}
 
+	public List<GeneratorTemplatesList> getModElementListTemplates(ModElement element) {
+		return getModElementListTemplates(element, false, null);
+	}
+
+	public List<GeneratorTemplatesList> getModElementListTemplates(ModElement element,
+			GeneratableElement generatableElement) {
+		return getModElementListTemplates(element, false, generatableElement);
+	}
+
+	private List<GeneratorTemplatesList> getModElementListTemplates(ModElement element, boolean performFSTasks,
+			GeneratableElement generatableElement) {
+
+		if (generatableElement == null) {
+			generatableElement = element.getGeneratableElement();
+			if (generatableElement == null) { // we can't construct list data because we have nothing to process
+				LOG.warn("Failed to load mod generatable element: " + element.getName()
+						+ ". This means no list templates will be generated");
+				return new ArrayList<>();
+			}
+		}
+
+		Map<?, ?> map = generatorConfiguration.getDefinitionsProvider().getModElementDefinition(element.getType());
+		if (map == null) {
+			LOG.info("Failed to load element definition for mod element type " + element.getType().getRegistryName());
+			return null;
+		}
+
+		Set<GeneratorTemplatesList> fileLists = new HashSet<>();
+
+		List<?> templateLists = (List<?>) map.get("list_templates");
+		if (templateLists != null) {
+			int templateID = 0;
+			int listID = 1;
+			for (Object list : templateLists) {
+				Map<ListTemplate, List<Boolean>> files = new LinkedHashMap<>();

```

Comment:
I think I asked for this before, but I still would like to see an explanation with comments why a list of booleans in a map is needed in the code, not just a silent code